src/commands/project/pipeline.ts

```diff
@@ -18,252 +24,180 @@ import {
+  };
 
-  if (missingConfig.length > 0) {
-    logger.error("Error in configuration: " + missingConfig.join(" "));
-    return false;
-  }
+  const map: { [key: string]: string | undefined } = {};
+  (Object.keys(values) as Array<keyof ICommandOptions>).forEach(key => {
+    const val = values[key];
+    if (key === "personalAccessToken") {
+      logger.debug(`${key}: XXXXXXXXXXXXXXXXX`);
+    } else {
+      logger.debug(`${key}: ${val}`);
+    }
+    map[key] = val;
+  });
 
-  return true;
+  const error = validateForRequiredValues(decorator, map);
+  return error.length > 0 ? null : values;
 };
 
 /**
- * Install the project hld lifecycle pipeline in an azure devops org.
+ * Executes the command.
  *
- * @param orgName
- * @param personalAccessToken
- * @param pipelineName
- * @param repositoryName
- * @param repositoryUrl
- * @param project
- * @param buildScriptUrl Build Script URL
- * @param exitFn
+ * @param opts Options object from commander.
+ * @param projectPath Project path which is the current directory.
+ * @param exitFn Exit function.
  */
-export const installLifecyclePipeline = async (
-  orgName: string,
-  personalAccessToken: string,
-  pipelineName: string,
-  repositoryName: string,
-  repositoryUrl: string,
-  project: string,
-  buildScriptUrl: string,
-  exitFn: (status: number) => void
+export const execute = async (
+  opts: ICommandOptions,
+  projectPath: string,
+  exitFn: (status: number) => Promise<void>
 ) => {
-  let devopsClient: IBuildApi | undefined;
-  let builtDefinition: BuildDefinition | undefined;
+  if (!projectPath) {
+    logger.error("Project Path is missing");
+    await exitFn(1);
+  } else {
+    logger.verbose(`project path: ${projectPath}`);
+
+    try {
+      await validate(projectPath);
+      const gitOriginUrl = await getOriginUrl();
+      const values = fetchValidateValues(opts, gitOriginUrl, Config());
 
-  try {
-    devopsClient = await getBuildApiClient(orgName, personalAccessToken);
-    logger.info("Fetched DevOps Client");
-  } catch (err) {
-    logger.error(err);
-    return exitFn(1);
+      if (values === null) {
+        await exitFn(1);
+      } else {
+        await installLifecyclePipeline(values);
+        await exitFn(0);
+      }
+    } catch (err) {
+      logger.error(
+        `Error occurred installing pipeline for HLD to Manifest pipeline`
+      );
+      logger.error(err);
+      await exitFn(1);
+    }
   }
+};
 
+export const commandDecorator = (command: commander.Command) => {
+  buildCmd(command, decorator).action(async (opts: ICommandOptions) => {
+    await execute(opts, process.cwd(), async (status: number) => {
+      await exitCmd(logger, process.exit, status);
+    });
+  });
+};
+
+const createPipeline = async (
+  values: ICommandOptions,
+  devopsClient: IBuildApi
+): Promise<BuildDefinition> => {
   const definition = definitionForAzureRepoPipeline({
-    branchFilters: ["master"],
+    branchFilters: ["master"], // TOFIX: hardcoded?
```

Comment:
I believe this is intentional. Our pipelines are triggered only by merges into the master branch.

@evanlouie @bnookala

Reply:
typically, when we hardcode a string like this. it is good to add some comments.

Reply:
Okay, then you can add the comment
> Pipelines are triggered only by merges into the master branch.

Reply:
will do. thanks